### PR TITLE
A small prototype of ellipse creation

### PR DESCRIPTION
**Pull request by DawnTheWitch**

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ */
+
+/**
+ * A type containing the X and Y position of a given point on a graph
+ * positionX The x position of this point
+ * positionY the y position of this point
+ */
+interface Point {
+    positionX: number;
+    positionY: number;
```
Comment (James-Oswald) [inline index.ts:13]: Simpler names here is fine, x and y, having to read 4x positionX and positionYs on one line is very verbose

File: index.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    if(radiusX < 0){
+        radiusX *= -1;
+    }
+    if (radiusY < 0){
+        radiusY *= -1;
+    }
+
+    ctx.beginPath();
+    ctx.ellipse(center.positionX, center.positionY, radiusX, radiusY, Math.PI / 2, 0, 2 * Math.PI); 
+    ctx.stroke();
```
Comment (James-Oswald) [inline index.ts:39]: This is good for the test, in the final application, you should move to the stroke call after you have drawn everything to save rendering time.

File: index.js
Comment (James-Oswald) [inline index.js:1]: Don't submit compiled .js files, add to .gitignore

**Review by James-Oswald — CHANGES_REQUESTED** [review]

File: index.ts (hunk 1 above)
Comment (DawnTheWitch) [reply index.ts:13]: I'll make that edit now, I figured it was bad to have such a short variable. Thank you

Comment (James-Oswald) [reply index.ts:13]: Short variable names ARE bad, but only if they're in a context where they can't be figured out at a glance. If a function is taking a point object and one sees `point.x` and `point.y`, everyone knows what these are without a more verbose name.

**Review by James-Oswald — COMMENTED** [review]
Good, only thing left is adding compiled js to .gitignore

**Review by James-Oswald — APPROVED** [review]

File: index.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    document.getElementById("ellipses")?.addEventListener("mousemove",mouseMoving);
+    document.getElementById("ellipses")?.addEventListener("mouseup",stopListening);
+}
+
+/**
+ * Follows the current mouse position, and draws the ellipse between the starting point and the current point.
+ * Clears the canvas with every movement.
+ * *When the tree is finished a redraw function will be made to draw all of the already created cuts and atoms.*
+ * @param event The even of a mouse moving
+ */
```
Comment (James-Oswald) [inline index.ts:64]: Keep lines, including comments, under 100 characters.

**Review by James-Oswald — APPROVED** [review]

File: index.ts (hunk 3 above)
Comment (DawnTheWitch) [reply index.ts:64]: Gotcha will fix and keep in mind!

**Pull request merged**